lib/Dialect/TritonGPU/Transforms/OptimizeThreadLocality.cpp

```diff
@@ -1,20 +1,95 @@
 #include "mlir/Pass/Pass.h"
 #include "mlir/Pass/PassManager.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
 #include "mlir/Transforms/Passes.h"
 #include "triton/Dialect/TritonGPU/IR/Dialect.h"
 #include "triton/Dialect/TritonGPU/Transforms/Passes.h"
 #include "triton/Dialect/TritonGPU/Transforms/Utility.h"
 #include <memory>
+#include <numeric>
 #define GEN_PASS_CLASSES
 #include "triton/Dialect/TritonGPU/Transforms/Passes.h.inc"
 
 using namespace mlir;
+namespace {
+// Change the layout of view destination when used by a reduction in order to
```

Comment:
AIUI reshape isn't the only view op, e.g. cat is another one.  This matcher seems only to be for reshapes, even though the name and variable names suggest it's more generic?

Reply:
(Indeed maybe it's clearer to make this name very specific, OptimizeReduceOfReshapeLayoutPattern?)

Reply:
Changed the name and the comments.